### ISSUE TITLE
[Regression]: Performance degradation in MNN 2.8.1 / 2.8.0 for Android CPU

### ISSUE THREAD

**Issue by Nick-infinity** [alibaba/MNN#2733]
# 平台(如果交叉编译请再附上交叉编译目标平台):  
# Platform(Include target platform as well if cross-compiling):  
Target : android 
Host: MacOS
NDK Version: 21,25

# Github版本:
# Github Version:

直接下载ZIP包请提供下载日期以及压缩包注释里的git版本(可通过``7z l zip包路径``命令并在输出信息中搜索``Comment`` 获得，形如``Comment = bc80b11110cd440aacdabbf59658d630527a7f2b``)。 git clone请提供 ``git commit`` 第一行的commit id

Provide date (or better yet, git revision from the comment section of the zip. Obtainable using ``7z l PATH/TO/ZIP`` and search for ``Comment`` in the output) if downloading source as zip,otherwise provide the first commit id from the output of ``git commit``

# 编译方式:
# Compiling Method
Cross compilation using android_build.sh (MNN-LLM)
```
请在这里粘贴cmake参数或使用的cmake脚本路径以及完整输出
Paste cmake arguments or path of the build script used here as well as the full log of the cmake proess here or pastebin
```
Build Script
https://github.com/wangzhaode/mnn-llm/blob/master/script/android_build.sh

# 编译日志:
# Build Log:

```
粘贴在这里
Paste log here or pastebin
```

MNN 2.8.0 Release  Specificaly mentions LLM performance of qwen1.8B model on android as 22t/s for decode. 
Neither the release assets libs nor the libs built from source can give the mentioned performance. 
I tried the Release Asset libraries as well as my source built libraries and I get around 7 t/s for qwen 1.8 Model.
(https://github.com/alibaba/MNN/releases/download/2.8.0/mnn_2.8.0_android_armv7_armv8_cpu_opencl_vulkan.zip)

Are there any specific flags , compiler options etc that I need to use to get the same performance ? or Are there any optimizations for which the code is not release by MNN team in Open Source Yet?
@MNNTeam 

another issue: https://github.com/wangzhaode/mnn-llm/issues/144

**Comment by v0jiuqi** [comment]
手机不同性能也不一样吧

**Comment by Nick-infinity** [comment]
> Different mobile phones have different performance, right?

Yes you are right, but I am testing with more powerful soc i.e 8 gen 2. 

The APK release on mnn-llm can do 26 t/s decode on my 8 gen 2 but the source built libs are doing 2 t/s . I found out that the mnn is not built with -DMNN_LOW_MEMORY cmake flag when using https://github.com/wangzhaode/mnn-llm/blob/master/script/android_build.sh

When I built the MNN libs with the low memory flag I could see a huge boost in performance. The performance is still not as good as the release apk though. I doubt that I am still missing some of the flags and options that were used to compile the mnn libs present in apk

**Comment by v0jiuqi** [comment]
测试大模型的时候需要手动添加这几个宏：
-DMNN_LOW_MEMORY=ON \
-DMNN_ARM82=ON 
只有打开了arm82，才会使用一些加速指令

**Comment by v0jiuqi** [comment]
另外，需要设置precision=low才会使用fp16推理，否则是fp32

**Comment by Nick-infinity** [comment]
Thank you very much for this information.
MNN_LOW_MEMORY this flag enables int4 weights support but activation are still in fp32. Will precision=low  make activations fp16? Also how can I enable precision=low

**Comment by v0jiuqi** [comment]
具体的一些设置（precision, threadNumber etc.）在 llm.cpp中，关于直接使用mnn来跑 llm ，可以打开宏 -DMNN_BUILD_LLM=ON，然后得到llm_demo这个可执行文件来跑大模型。

**Comment by Nick-infinity** [comment]
Thank you very much @v0jiuqi . You have saved my day and helped me to fix the issue. I can confirm I can get the same perf now as the released apk. 
I cant thank you enough <3 . 
Thanks again and have a good day
